modules/core/src/rand.cpp

```diff
@@ -511,8 +391,8 @@ static RandnScaleFunc randnScaleTab[] =
 void RNG::fill( InputOutputArray _mat, int disttype,
                 InputArray _param1arg, InputArray _param2arg, bool saturateRange )
 {
-    CV_Assert(!_mat.empty());
-
+    if (_mat.empty())
+        return;
```

Comment:
Early error detection: #8300

Filling **empty** Mat with random numbers is not a normal use case of OpenCV.